contracts/Strategy.sol

```diff
@@ -101,21 +122,32 @@ contract Strategy is BaseStrategy {
     }
 
     function _claimAndSellRewards() internal {
-        if (!bypassClaim) rewardDistributor.claim();
+        if (!bypassClaim && rewardDistributor.getClaimable(address(this)) > 0) {
+            rewardDistributor.claim();
+        }
 
         SwapThresholds memory st = swapThresholds;
         uint256 rewardBalance = balanceOfReward();
         if (rewardBalance > st.min) {
             // Redeem the full balance at once to avoid unnecessary costly withdrawals.
-            IERC4626(address(rewardToken)).redeem(rewardBalance, address(this), address(this));
+            uint256 output = IERC4626(address(rewardToken)).redeem(
+                rewardBalance,
+                address(this),
+                address(this)
+            );
+            
+            if (st.autoAdjustThresholds) {
+                // use our weekly output to set how much we max sell each time (make sure we get it all in 7 days)
+                swapThresholds.max = uint112((output * 101) / 700);
+            }
         }
 
         uint256 toSwap = rewardTokenUnderlying.balanceOf(address(this));
         if (toSwap > st.min) {
             toSwap = min(toSwap, st.max);
```

Comment:
`st.max` may be using a stale value because it is using the memory value set on line 129 and not the updated value that was set into storage on 141.

suggestion, add a line above 141 to update the memory value (`st.max`) as well.